### PR TITLE
Start integration tests with clean state in Anvil and DB, stop everything after integration tests are done

### PR DESCRIPTION
**Pull request by antonyshchenko**

### CODE REVIEW

File: .github/workflows/clear-database.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+      - name: Configure SSH Agent
+        uses: webfactory/ssh-agent@v0.9.0
+        with:
+          ssh-private-key: ${{ secrets.CI_BASTION_KEY }}
+      - name: Clear database
+        run: |
+          ssh -o StrictHostKeyChecking=no -L 5434:test-db-cluster.cluster-cpwwaa4iqa1s.us-east-2.rds.amazonaws.com:5432 ec2-user@test-bastion.funkybit.fun -N &
+          sleep 2  # to give the SSH tunnel time to be set up
+          PGPASSWORD=$(aws --region us-east-2 secretsmanager get-secret-value --secret-id 'arn:aws:secretsmanager:us-east-2:851725450525:secret:rds!cluster-a00059f0-7cd7-4989-bd67-7af9e5899397-cWukH6' | jq -r '.SecretString' | jq -r '.password') \
+            psql --port=5434 --host=localhost --user=funkybit -c "DROP SCHEMA IF EXISTS public CASCADE; CREATE SCHEMA public;"
```
Comment (bholzman) [inline .github/workflows/clear-database.yml:33]: I usually do `dropdb --port=5434 --host=localhost --user=funkybit funkybit` and then `createdb --port=5434 --host=localhost --user=funkybit funkybit` but I guess this is just as good

**Review by bholzman — APPROVED** [review]
Looks good!

File: .github/workflows/CI.yml [hunk 2]
```diff
@@ -28,29 +28,41 @@ jobs:
...
-    uses: ./.github/workflows/deploy-backend.yml
+    uses: ./.github/workflows/upgrade-services.yml
     with:
-      action: upgrade
       environment: test
+      services: "api,ring,telegrambot,garp,sequencer"
       tag: ${{ needs.build_push_backend.outputs.docker_image_tag }}
 
+  start_services:
+    name: Stop Backend
```
Comment (bflood366) [inline .github/workflows/CI.yml:55]: should be `Start Backend` or `Start services`

File: .github/workflows/CI.yml [hunk 3]
```diff
@@ -75,11 +87,24 @@ jobs:
...
     permissions:
       id-token: write # needed for aws credentials
       contents: read
       checks: write # for writing test reports
     uses: ./.github/workflows/integration-tests.yml
     secrets: inherit # needs access to secrets.CI_BASTION_KEY
+
+  stop_services:
+    name: Stop services
+    if: ${{ github.event_name == 'push' && always() }}
```
Comment (bflood366) [inline .github/workflows/CI.yml:101]: whats does `&& always` do here

File: .github/workflows/CI.yml [hunk 4]
```diff
@@ -28,29 +28,41 @@ jobs:
...
+    needs: [ clear_database ]
     permissions:
       id-token: write # needed for aws credentials
       contents: read
-    uses: ./.github/workflows/deploy-backend.yml
+    uses: ./.github/workflows/upgrade-services.yml
     with:
-      action: upgrade
       environment: test
+      services: "api,ring,telegrambot,garp,sequencer"
```
Comment (bflood366) [inline .github/workflows/CI.yml:51]: it seems like we should start ring first since I believe that is the one that does the db migration.

**Review by bflood366 — APPROVED** [review]
looks good to me

File: .github/workflows/CI.yml (hunk 2 above)
Comment (antonyshchenko) [reply .github/workflows/CI.yml:55]: 👍

File: .github/workflows/CI.yml (hunk 3 above)
Comment (antonyshchenko) [reply .github/workflows/CI.yml:101]: `&& always()` means that this job would run even if `integration_tests` job has failed or workflow was canceled.

Comment (antonyshchenko) [reply .github/workflows/CI.yml:101]: I will add a comment on this

**Review by imykolenko — COMMENTED** [review]
Nice cleanup!

**Pull request merged**